src/tests/nativeaot/SmokeTests/PInvoke/PInvoke.cs

```diff
@@ -1102,6 +1109,31 @@ public static unsafe void TestForwardDelegateWithUnmanagedCallersOnly()
             Action a = Marshal.GetDelegateForFunctionPointer<Action>((IntPtr)(void*)(delegate* unmanaged<void>)&UnmanagedCallback);
             a();
         }
+		
+        public static unsafe void TestDifferentModopts()
+		{
+			byte storage;
+
+			delegate* unmanaged<byte*, byte, void> unmanagedMethod = &UnmanagedMethod;
+
+			var outUnmanagedMethod = (delegate* unmanaged<out byte, byte, void>)unmanagedMethod;
+			outUnmanagedMethod(out storage, 12);
+			ThrowIfNotEquals(storage, 12, "Out unmanaged call failed.");
+
+			var refUnmanagedMethod = (delegate* unmanaged<ref byte, byte, void>)unmanagedMethod;
+			refUnmanagedMethod(ref storage, 34);
+			ThrowIfNotEquals(storage, 34, "Ref unmanaged call failed.");
+
+			delegate* unmanaged[Stdcall]<byte*, byte, void> stdcallMethod = &StdcallMethod;
+
+			var outStdcallMethod = (delegate* unmanaged[Stdcall]<out byte, byte, void>)stdcallMethod;
+			outStdcallMethod(out storage, 12);
+			ThrowIfNotEquals(storage, 12, "Out unmanaged stdcall failed.");
+
+			var refStdcallMethod = (delegate* unmanaged[Stdcall]<ref byte, byte, void>)stdcallMethod;
+			refStdcallMethod(ref storage, 34);
+			ThrowIfNotEquals(storage, 34, "Ref unmanaged stdcall failed.");
+		}
```

Comment:
Add this to make the compiler assert.

We also need to fix up the way these sort.

```suggestion
            var refStdcallSuppressTransition = (delegate* unmanaged[Stdcall, SuppressGCTransition]<ref byte, byte, void>)stdcallMethod;
            if (string.Empty.Length > 0)
            {
                // Do not actually call this because the calling convention is wrong. We just check the compiler didn't crash.
                refStdcallSuppressTransition(ref storage, 56);
            }
		}
```